src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseWeakAsymmetricAlgorithm.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotUseWeakAsymmetricAlgorithm : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5384";
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotUseWeakAsymmetricAlgorithm),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotUseWeakAsymmetricAlgorithmMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotUseWeakAsymmetricAlgorithmDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+
+        private static readonly ImmutableHashSet<string> s_RSAAlgorithmNames =
+            ImmutableHashSet.Create(
+                "RSA",
+                "System.Security.Cryptography.RSA",
+                "System.Security.Cryptography.AsymmetricAlgorithm");
+
+        private static readonly ImmutableHashSet<string> s_DSAAlgorithmNames =
```

Comment:
>s_DSAAlgorithmNames [](start = 57, length = 19)

I think it might be worth separating out different rules for RSA and DSA, since the guidance will probably be different (e.g. RSA -> use a big key size if possible, DSA -> use something else if possible)